src/main.rs

```diff
@@ -869,4 +913,188 @@ mod tests {
+            conf: &sgx_validation::config::Config,
+        ) -> anyhow::Result<()> {
+            let sgx = sgx_validation::Sgx::default();
+
+            #[allow(unused_variables)]
+            for Attribute { oid, values } in csr.info.attributes.iter() {
+                for any in values.iter() {
+                    let ereq: ExtensionReq<'_> = any.decode_into()?;
+                    for ext in Vec::from(ereq) {
+                        let (quote, bytes): (Quote<'_>, _) = ext.extn_value.parse()?;
+                        let chain = quote.chain()?;
+                        let chain = chain
+                            .iter()
+                            .map(|c| Certificate::from_der(c))
+                            .collect::<Result<Vec<_>, _>>()?;
+
+                        // Validate the report.
+                        let pck = sgx.trusted(&chain)?;
+                        let report = quote.verify(pck)?;
+                        sgx.verify(&csr.info, &ext, Some(conf), false)?;
+                    }
+                }
+            }
+            Ok(())
+        }
+
+        fn assert_snp_config(
+            csr: &CertReq<'_>,
+            conf: &snp_validation::config::Config,
+        ) -> anyhow::Result<()> {
+            let snp = Snp::default();
+            #[allow(unused_variables)]
+            for Attribute { oid, values } in csr.info.attributes.iter() {
+                for any in values.iter() {
+                    let ereq: ExtensionReq<'_> = any.decode_into()?;
+                    for ext in Vec::from(ereq) {
+                        let evidence = Evidence::from_der(ext.extn_value)?;
+                        let array = evidence.report.try_into()?;
+                        let report = Report::cast(array);
+                        snp.verify(&csr.info, &ext, Some(conf), false)?;
+                    }
+                }
+            }
+            Ok(())
+        }
+
+        #[test]
+        fn test_config_empty() {
+            let config: Config = toml::from_str("").expect("Couldn't deserialize");
+
+            assert!(config.sgx.is_none());
+            assert!(config.snp.is_none());
+        }
+
+        #[test]
+        fn test_config() {
+            let config: Config = toml::from_str(
+                r#"
+            [snp]
+            policy_flags = ["SingleSocket", "Debug", "SMT"]
+            enarx_signer = ["e368c18e60842db9325778532dd81594d732078bf01aa91686be40333da639e08733b910bd057bdda50d715968b075ce"]
+            abi = ">1.0"
+            
+            [sgx]
+            enarx_signer = ["2eba0f494f428e799c22d6f12778aebea4dc8d991f9e63fd3cddd57ac6eb5dd9"]
+            "#,
+            )
+            .expect("Couldn't deserialize");
+
+            assert!(config.sgx.is_some());
+            assert!(config.snp.is_some());
+            assert_eq!(
+                config.sgx.unwrap().measurements.signer.len(),
+                config.snp.unwrap().measurements.signer.len()
+            );
```

Comment:
If you want to test the config parsing, then use a single `assert_eq!` with the whole config.

I would expect to see no isolated config parsing tests here at all, I'd expect to see Steward initialized with a config and handling real requests

Reply:
Done